examples/opensora_pku/opensora/train/train_t2v.py

```diff
@@ -248,32 +263,100 @@ def main(args):
     assert dataset_size > 0, "Incorrect dataset size. Please check your dataset size and your global batch size"
 
     # 4. build training utils: lr, optim, callbacks, trainer
-    if args.max_train_steps is not None and args.max_train_steps > 0:
-        args.epochs = args.max_train_steps // dataset_size
-        logger.info(f"Forcing training epochs to {args.epochs} when using max_train_steps {args.max_train_steps}")
-    if args.checkpointing_steps is not None and args.checkpointing_steps > 0:
-        logger.info(f"Saving checkpoints every {args.checkpointing_steps} steps")
-        args.step_mode = True
-        args.ckpt_save_interval = args.checkpointing_steps
+    if args.scale_lr:
+        learning_rate = args.start_learning_rate * args.batch_size * args.gradient_accumulation_steps * device_num
+        end_learning_rate = args.end_learning_rate * args.batch_size * args.gradient_accumulation_steps * device_num
+    else:
+        learning_rate = args.start_learning_rate
+        end_learning_rate = args.end_learning_rate
+
+    if args.dataset_sink_mode and args.sink_size != -1:
+        assert args.sink_size > 0, f"Expect that sink size is a positive integer, but got {args.sink_size}"
+        steps_per_sink = args.sink_size
+    else:
+        steps_per_sink = dataset_size
+
+    if args.max_train_steps is not None:
+        assert args.max_train_steps > 0, f"max_train_steps should a positive integer, but got {args.max_train_steps}"
+        assert (
+            args.max_train_steps >= steps_per_sink
+        ), f"Expect that the max_train_steps is no less than {steps_per_sink}, but got {args.max_train_steps}"
+        total_train_steps = args.max_train_steps
+        args.epochs = total_train_steps // dataset_size
+    else:
+        # use args.epochs
+        assert (
+            args.epochs is not None and args.epochs > 0
+        ), f"When args.max_train_steps is not provided, args.epochs must be a positive integer! but got {args.epochs}"
+        total_train_steps = args.epochs * dataset_size
+
+    sink_epochs = total_train_steps // steps_per_sink
```

Comment:
sink_epochs = math.ceil(total_train_steps / steps_per_sink)

Reply:
`max_train_steps` is the **maximum** training steps, meaning `total_train_steps <=max_train_steps`. Therefore the `sink_epochs` should be the floor not the ceiling of `max_train_steps/step_per_sink`. 

To avoid the `sink_epochs` being zero, there is an assertion:
```python
        assert (
            args.max_train_steps >= steps_per_sink
        ), f"Expect that the max_train_steps is no less than {steps_per_sink}, but got {args.max_train_steps}"
        total_train_steps = args.max_train_steps
        args.epochs = total_train_steps // dataset_size
```
If the user gives too small `args.max_train_steps`, the program will raise an error, and ask the user to increase `args.max_train_steps` to the value of `steps_per_sink`.
I will further improve the printing message.

Reply:
Fixed it according to your suggestion.